gateway/header.go

```diff
@@ -113,3 +113,19 @@ func handleForwardResponseTrailer(w http.ResponseWriter, md runtime.ServerMetada
 		}
 	}
 }
+
+// CustomIncomingHeaderMatcher func is used to add custom headers to be matched
+// from incoming http requests, If this returns true the header will be added to grpc context
+func CustomIncomingHeaderMatcher(headerNames ...string) func(string) (string, bool) {
```

Comment:
Not sure I like the name of this. It could be used for either incoming or outgoing headers, so `Incoming` can be removed. And nothing about the name implies that it's additive with the default header matcher. The description is good, maybe name it like it is described:
```suggestion
func AdditionalHeaderMatcher(headerNames ...string) func(string) (string, bool) {
```